axelor-base/src/main/java/com/axelor/apps/base/service/pricing/PricingComputer.java

```diff
@@ -146,7 +146,13 @@ public void apply() throws AxelorException {
     LOG.debug("Treating pricing childs of {}", this.pricing);
     for (int counter = 0; counter < MAX_ITERATION; counter++) {
 
-      Optional<Pricing> optChildPricing = getNextPricing(currentPricing);
+      Optional<Pricing> optChildPricing = Optional.empty();
+
+      if (pricing.getLinkedPricing() == null) {
```

Comment:
I might be wrong, but is there a infinite loop for case with boolean is switched for"previousPricing" ?
**First iteration** for this case is ok, since linkedPricing of the root will be null, so for next iteration it will fetch pricing where linkedPricing = currentPricing (so its child) and then start second iteration with it
**On second iteration**, linkedPricing will not be null (pointed to the pricing of the first iteration because it is its parent) and so nextPricing will actually be determined as its current parent, repeating then the first iteration

A possible solution would be to add a boolean in init of PricingComputer (to know wether its "previous" or "next" mode) and then depending on the value, in PricingComputer you can either call `getPreviousPricing` or `Optional.ofNullable(currentPricing.getLinkedPricing())` to get the next pricing. (You can even keep the commented code above but you will have to add condition)

Reply:
@kra-axelor 
so i don't think it will be infinite looping

Reply:
Aah yes indeed, you are right